include/ceed.h

```diff
@@ -242,11 +242,28 @@ typedef enum {
   CEED_GAUSS_LOBATTO = 1,
 } CeedQuadMode;
 
+/// Type of basis shape to create non-tensor H1 element basis
+/// @ingrounp CeedBasis
+typedef enum {
+  /// Quad/Hex/Hypercube - 1+D shape
+  CEED_HCUBE = 0,
+  /// Simplex/tet - 2+D shape
+  CEED_SIMPLEX = 1,
+  /// Pyramid - 3D shape
+  CEED_PRYAMID = 2,
+  /// Prism - 3D shape
+  CEED_PRISM = 4,
+} CeedElemTopology;
```

Comment:
I only added tests for 'CEED_SIMPLEX' right now, but for the ref backend it should all be the same. I'm not sure if other backends would want to know what type of non-tensor basis is being applied, but that is easy to add to the private basis data if desirable.

Reply:
So enumerating CEED_LINE, CEED_SQUARE, CEED_CUBE, CEED_TRIANGLE, CEED_TETRAHEDRA separately?

Reply:
If you're handling topology in a switch statement, it's easier for it to be flat. Also, we'll need to classify topology (to have homogeneous batches) and the key is simpler when it's one thing rather than two.